cmd/rest/client.go

```diff
@@ -42,15 +42,17 @@ type Client struct {
 }
 
 // Call - make a REST call.
-func (c *Client) Call(method string, values url.Values, body io.Reader) (reply io.ReadCloser, err error) {
+func (c *Client) Call(method string, values url.Values, body io.Reader, length int64) (reply io.ReadCloser, err error) {
 	req, err := http.NewRequest(http.MethodPost, c.url.String()+"/"+method+"?"+values.Encode(), body)
 	if err != nil {
 		return nil, err
 	}
 
 	req.Header.Set("Authorization", "Bearer "+c.newAuthToken())
 	req.Header.Set("X-Minio-Time", time.Now().UTC().Format(time.RFC3339))
-
+	if length != -1 {
```

Comment:
this code means that if `length is -2` then it can be set as ContentLength. You should simply do `length > 0`